### ISSUE TITLE
Activity is not loading when configured to show "A file or folder has been changed or renamed"

### ISSUE THREAD

**Issue by Nox996** [nextcloud/activity#486]
### Steps to reproduce (I can reproduce this only for one user)
1. Installed Nextcloud using docker-compose (with-nginx-proxy mariadb-cron-redis fpm from nextcloud docker github)
2. Create users and log in with one of them
3. Enable TOTP two factor authentication
4. Copy / move some files and folders
5. Click "Activity"

Note: Mindmap is *not* installed, the installation is completely fresh.

### Expected behaviour
Activities should be listed

### Actual behaviour
Activities hangs / keeps loading forever.

**Note: If "A file or folder has been changed or renamed" is disabled in Settings for Activity app, the Activities of the past load flawlessly.**

### Server configuration

Nextcloud was installed using docker-compose and https://github.com/nextcloud/docker/tree/master/.examples/docker-compose/with-nginx-proxy/mariadb-cron-redis/fpm
**Operating system**: Arch Linux / docker-compose

**Web server:** nginx-proxy / fpm

**Database:** mariadb

**PHP version:** php7.4

**Nextcloud version:** (see Nextcloud admin page) 19.0.1

**Where did you install Nextcloud from:** https://github.com/nextcloud/docker/tree/master/.examples/docker-compose/with-nginx-proxy/mariadb-cron-redis/fpm

**Signing status:** no modifications have been made

```
Login as admin user into your Nextcloud and access
http://example.com/index.php/settings/integrity/failed
paste the results here.

No errors have been found.
```

**List of activated apps:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```
Enabled:
  - accessibility: 1.5.0
  - activity: 2.12.0
  - bruteforcesettings: 1.6.0
  - cloud_federation_api: 1.2.0
  - comments: 1.9.0
  - contactsinteraction: 1.0.0
  - dav: 1.15.0
  - federatedfilesharing: 1.9.0
  - federation: 1.9.0
  - files: 1.14.0
  - files_pdfviewer: 1.8.0
  - files_rightclick: 0.16.0
  - files_sharing: 1.11.0
  - files_trashbin: 1.9.0
  - files_versions: 1.12.0
  - files_videoplayer: 1.8.0
  - firstrunwizard: 2.8.0
  - logreader: 2.4.0
  - lookup_server_connector: 1.7.0
  - nextcloud_announcements: 1.8.0
  - notifications: 2.7.0
  - oauth2: 1.7.0
  - password_policy: 1.9.1
  - photos: 1.1.0
  - privacy: 1.3.0
  - provisioning_api: 1.9.0
  - recommendations: 0.7.0
  - serverinfo: 1.9.0
  - settings: 1.1.0
  - sharebymail: 1.9.0
  - support: 1.2.1
  - survey_client: 1.7.0
  - systemtags: 1.9.0
  - text: 3.0.1
  - theming: 1.10.0
  - twofactor_backupcodes: 1.8.0
  - twofactor_totp: 4.1.3
  - updatenotification: 1.9.0
  - viewer: 1.3.0
  - workflowengine: 2.1.0
Disabled:
  - admin_audit
  - encryption
  - files_external
  - user_ldap

**Nextcloud configuration:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

{
    "system": {
        "memcache.local": "\\OC\\Memcache\\APCu",
        "apps_paths": [
            {
                "path": "\/var\/www\/html\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/custom_apps",
                "url": "\/custom_apps",
                "writable": true
            }
        ],
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "nc.bruchtal.dynu.net"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "19.0.1.1",
        "overwrite.cli.url": "http:\/\/nc.bruchtal.dynu.net",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "overwriteprotocol": "https",
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "tls",
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "password": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "app_install_overwrite": [
            "issuetemplate"
        ]
    }
}
```

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
none
### Client configuration
**Browser:**
Chromium / Chrome / Firefox, no difference
**Operating system:**
Arch Linux, Windows 10, no difference
#### Nextcloud log (data/nextcloud.log)
Insert your Nextcloud log here
[nextcloud.log](https://github.com/nextcloud/activity/files/5112989/nextcloud.log)

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log:
```
```
23:02:04.952 JQMIGRATE: Migrate is installed, version 1.4.1 jquery-migrate.min.js:2:551
23:02:04.996 $ is deprecated: The global jQuery is deprecated. It will be updated to v2.4 in Nextcloud 20 and v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own. globals.js:61:15
23:02:05.010 Handlebars is deprecated: please ship your own, this will be removed in Nextcloud 20 globals.js:61:15
23:02:05.023 $ is deprecated: The global jQuery is deprecated. It will be updated to v2.4 in Nextcloud 20 and v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own. 3 globals.js:61:15
23:02:05.024 Notifications permissions denied App.vue:296
23:02:05.035 $ is deprecated: The global jQuery is deprecated. It will be updated to v2.4 in Nextcloud 20 and v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own. globals.js:61:15
23:02:05.038 jQuery is deprecated: The global jQuery is deprecated. It will be updated to v2.4 in Nextcloud 20 and v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own. globals.js:61:15
23:02:05.042 Handlebars is deprecated: please ship your own, this will be removed in Nextcloud 20 globals.js:61:15
23:02:05.043 $ is deprecated: The global jQuery is deprecated. It will be updated to v2.4 in Nextcloud 20 and v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own. 2 globals.js:61:15
23:02:05.066 session heartbeat polling started session-heartbeat.js:97:9
23:02:05.082 $ is deprecated: The global jQuery is deprecated. It will be updated to v2.4 in Nextcloud 20 and v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own. 11 globals.js:61:15
23:02:05.101 Clipboard is deprecated: please ship your own, this will be removed in Nextcloud 20 globals.js:61:15
23:02:05.102 $ is deprecated: The global jQuery is deprecated. It will be updated to v2.4 in Nextcloud 20 and v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own. 2 globals.js:61:15
23:02:05.126 XHRGEThttps://nc.bruchtal.dynu.net/ocs/v2.php/apps/activity/api/v2/activity/all?format=json&previews=true&since=0
[HTTP/2 404 Not Found 116ms]
```

23:02:05.230 $ is deprecated: The global jQuery is deprecated. It will be updated to v2.4 in Nextcloud 20 and v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own. globals.js:61:15

b) The network log
c) ...

**Comment by Castiontone** [comment]
I can confirm the part 'Activities hangs / keeps loading forever. Note: If "A file or folder has been changed or renamed" is disabled in Settings for Activity app, the Activities of the past load flawlessly.'

Setup of server is different: apache2, ubuntu 18.04, php7.4, mysql

**Comment by OrelEagle** [comment]
I started having this issue after editing a mindmap file (.mm extension) on my computer inside a synchronized folder. The file had been there for a long time but it was the first time I edited it after upload. 
Issue temporarily fixed by unchecking "A file or folder has been changed or renamed" and "A file or folder has been deleted", as I deleted this file later.

I'm also getting this error in the logs: "Exception: substr() expects parameter 3 to be int, bool given", referring to the mindmap mime type:

```
/var/www/nextcloud/apps/activity/lib/Controller/APIv2.php - line 408:
OC\Files\Type\Detection->mimeTypeIcon("mindmap")
/var/www/nextcloud/apps/activity/lib/Controller/APIv2.php - line 399:
OCA\Activity\Controller\APIv2->getPreviewPathFromMimeType("application/x-freemind")
```
Note: I didn't install the mindmap app!
